app/models/timestamp.rb

```diff
@@ -53,8 +55,12 @@ def parse!
         ActiveSupport::Duration.parse(@timestamp_string).iso8601
       when DATE_KEYWORD_REGEX # Built in date keywords
         @timestamp_string
+      when DATE_REGEX
+        # While converting a date string "2023-01-01" to an iso8601 time object, use the
+        # system's Time.zone timezone as a reference.
+        Date.iso8601(@timestamp_string).in_time_zone(Time.zone).iso8601
       else
-        Time.zone.iso8601(@timestamp_string).iso8601
+        Time.iso8601(@timestamp_string).iso8601
```

Comment:
```suggestion
        Time.parse(@timestamp_string).iso8601
```

This fails when providing the values from the frontend, which currently do not output seconds (but simply HH:mm).

```
Time.iso8601("2022-10-29T21:55+03:00")
ArgumentError: invalid xmlschema format: "2022-10-29T21:55+03:00
```

`Time.parse` works however

Reply:
Fixed the issue, `Time.parse` is too permissive and it parsed values that are supposed to be invalid. I used the `DateTime.iso8601` as it is more compatible with the iso standard, than `Time.iso8601`. One caveat is that, when a UTC timestamp "2023-01-01T00:00:00Z" is provided, it will parse it as "2023-01-01T00:00:00+00:00" adding the timezone designator, but we can live with that.